pytest_httpserver/blocking_http_server.py

```diff
@@ -0,0 +1,155 @@
+from queue import Empty
+from queue import Queue
+from typing import Any
+from typing import Dict
+from typing import Mapping
+from typing import Optional
+from typing import Pattern
+from typing import Union
+
+from werkzeug.wrappers import Request
+from werkzeug.wrappers import Response
+
+from pytest_httpserver.httpserver import METHOD_ALL
+from pytest_httpserver.httpserver import UNDEFINED
+from pytest_httpserver.httpserver import HeaderValueMatcher
+from pytest_httpserver.httpserver import HttpServerBase
+from pytest_httpserver.httpserver import QueryMatcher
+from pytest_httpserver.httpserver import RequestHandlerBase
+from pytest_httpserver.httpserver import URIPattern
+
+
+class BlockingRequestHandler(RequestHandlerBase):
+    """
+    Provides responding to a request synchronously.
+
+    This class should only be instantiated inside the implementation of the :py:class:`BlockingHttpServer`.
+    """
+
+    def __init__(self):
+        self.response_queue = Queue()
+
+    def respond_with_response(self, response: Response):
+        self.response_queue.put_nowait(response)
+
+
+class BlockingHttpServer(HttpServerBase):
+    """
+    Server instance which enables synchronous matching for incoming requests.
+
+    :param timeout: waiting time in seconds for matching and responding to an incoming request.
+        manager
+
+    For further parameters and attributes see :py:class:`HttpServerBase`.
+    """
+
+    def __init__(self, timeout: int = 30, **kwargs):
```

Comment:
I think we should copy kwargs to here from the base class and avoid using `**kwargs`. This helps IDEs a lot and also makes the documentation better.
I know it is copy&paste but I have no better idea, but it is still better than `**kwargs`.

Reply:
Let's do it.